pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscLowerOperatorOp.cpp

```diff
@@ -0,0 +1,160 @@
+  LogicalResult matchAndRewrite(
+      OperatorOp op,
+      OpAdaptor adaptor,
+      ConversionPatternRewriter& rewriter) const override {
+    Location loc = op.getLoc();
+    auto name = op.name();
+    if ("torch_blade.fake_quant" == name) {
+      auto operands = adaptor.operands();
+      auto input = operands[0];
+      auto scale = operands[1];
+      auto zeroPoint = operands[2];
+
+#define I64_VAR_FROM_CONST_OPERAND(var, idx)                         \
+  int64_t var;                                                       \
+  if (!matchPattern(op.operands()[idx], m_TorchConstantInt(&var))) { \
+    return op.emitError(#var " must be a Scalar constant int");      \
+  }
+      I64_VAR_FROM_CONST_OPERAND(qmin, 3);
+      I64_VAR_FROM_CONST_OPERAND(qmax, 4);
+      I64_VAR_FROM_CONST_OPERAND(numBits, 5);
+#undef I64_VAR_FROM_CONST_OPERAND
+
+      SmallVector<int64_t, 4> axis;
+      if (!matchPattern(op.operands()[6], m_TorchConstantIntList(axis))) {
+        return op.emitError("Only constant dims are currently supported");
+      }
+
+#define BOOL_VAR_FROM_CONST_OPERAND(var, idx)                         \
+  bool var;                                                           \
+  if (!matchPattern(op.operands()[idx], m_TorchConstantBool(&var))) { \
+    return op.emitError(#var " must be a Scalar constant boolean");   \
+  }
+
+      BOOL_VAR_FROM_CONST_OPERAND(useSigned, 7);
+      BOOL_VAR_FROM_CONST_OPERAND(useSymmetric, 8);
+      BOOL_VAR_FROM_CONST_OPERAND(useDynamic, 9);
+      BOOL_VAR_FROM_CONST_OPERAND(usePerChannel, 10);
+#undef BOOL_VAR_FROM_CONST_OPERAND
+
+      auto resultTy = input.getType().dyn_cast<RankedTensorType>();
+      if (!resultTy) {
+        return op.emitError("failed to get type of input");
+      }
+      auto castedResultTy =
+          RankedTensorType::get(resultTy.getShape(), rewriter.getF32Type());
+
+      auto zeroPointTy = zeroPoint.getType().dyn_cast<RankedTensorType>();
+      if (!zeroPointTy) {
+        return op.emitError("zero point is not a RankedTensorType");
+      }
+      auto i32Ty = rewriter.getIntegerType(32);
+      auto i64Ty = rewriter.getIntegerType(64);
+      auto castedZeroPointTy =
+          RankedTensorType::get(zeroPointTy.getShape(), i32Ty);
+      Value castedZeroPoint =
+          rewriter.create<mhlo::ConvertOp>(loc, castedZeroPointTy, zeroPoint);
+      // Attributes ...
+      auto axisAttr = DenseIntElementsAttr::get(
+          RankedTensorType::get({axis.size()}, rewriter.getIntegerType(64)),
+          axis);
+      auto numBitsAttr = rewriter.getIntegerAttr(i64Ty, numBits);
+      auto qminAttr = rewriter.getIntegerAttr(i64Ty, qmin);
+      auto qmaxAttr = rewriter.getIntegerAttr(i64Ty, qmax);
+      auto useSignedAttr = rewriter.getBoolAttr(useSigned);
+      auto useSymmetricAttr = rewriter.getBoolAttr(useSymmetric);
+      auto useDynamicAttr = rewriter.getBoolAttr(useDynamic);
+      Value newOp = rewriter.create<mhlo_disc::FakeQuantOp>(
```

Comment:
Nit: use rewriter.replaceOpWithNewOp

Reply:
Any tricks about it?

Reply:
It seems `rewriter.replaceOpWithNewOp ` makes one line code instead of two lines.

Reply:
Well... it's 13 v.s 14 LOC, not much difference.

Reply:
`replaceOpWithNewOp` is more intuitive, and we should use the return type from op.getType().

Reply:
Well, in this conversion pattern, return type of converted op (saying `mhlo_disc::FakeQuantOp`) may NOT be same as the original op (saying `torch.OperatorOp`), so the return type should be explicitly constructed to be legal type in converted dialect. More specifically in this case, operand `input` and result share the same data type, so type of `input`, which is already converted by `OpAdaptor` is reused.